flake.nix

```diff
@@ -0,0 +1,69 @@
+{
+  description = "Apple M1 support for NixOS";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+  };
+
+  outputs = { self, flake-parts, ... }@inputs:
+    flake-parts.lib.mkFlake { inherit inputs; } (
+      { withSystem, ... }: {
+        flake = {
+          overlays = rec {
+            asahi-overlay = import packages/overlay.nix;
+            default = asahi-overlay;
+          };
+
+          nixosModules = rec {
+            m1-support = ./nixos-module;
+            default = m1-support;
+          };
+        };
+
+        # build platforms supported for uboot in nixpkgs
+        systems = [ "aarch64-linux" "x86_64-linux" "i686-linux" ];
```

Comment:
We can drop `i686-linux` here. The comment is useful though, if it does not evaluate properly on other systems.